cli/flox/src/utils/init/logger.rs

```diff
@@ -51,29 +51,22 @@ static LOGGER_HANDLE: OnceCell<(
     >,
 )> = OnceCell::new();
 
-pub fn init_logger(verbosity: Option<Verbosity>, debug: Option<bool>) {
+pub fn init_logger(verbosity: Option<Verbosity>, debug: bool) {
     let verbosity = verbosity.unwrap_or_default();
-    let debug = debug.unwrap_or(false);
+    let debug = debug || matches!(verbosity, Verbosity::Verbose(1..));
 
-    let log_filter = match (debug, verbosity) {
+    let log_filter = match verbosity {
         // Show only errors
-        (false, Verbosity::Quiet) => "off,flox=error",
+        Verbosity::Quiet => "off,flox=error",
         // Show our own info logs
-        (false, Verbosity::Verbose(0)) => "off,flox=info",
-        // Also show POSIX info
-        (false, Verbosity::Verbose(1)) => "off,flox=info,posix=info",
-        // Also show info from our libraries and POSIX debug
-        (false, Verbosity::Verbose(2)) => {
-            "off,flox=debug,flox-rust-sdk=info,runix=info,posix=debug"
-        },
+        Verbosity::Verbose(0) => "off,flox=info",
         // Also show debug from our libraries
-        (true, Verbosity::Quiet) | (false, Verbosity::Verbose(3)) => {
-            "off,flox=debug,flox-rust-sdk=debug,runix=debug,posix=debug"
-        },
-        // Also show debug from everything
-        (true, Verbosity::Verbose(0)) | (false, Verbosity::Verbose(4)) => "debug",
-        // Also show trace from everything
-        (true, Verbosity::Verbose(_)) | (false, Verbosity::Verbose(_)) => "trace",
+        Verbosity::Verbose(1) => "off,flox=debug,flox-rust-sdk=debug,runix=debug",
+        // Also show trace from our libraries and POSIX
+        Verbosity::Verbose(2) => "off,flox=trace,flox-rust-sdk=trace,runix=trace",
+        // Also show trace from our libraries and POSIX
+        Verbosity::Verbose(3) => "debug,flox=trace,flox-rust-sdk=trace,runix=trace",
+        Verbosity::Verbose(_) => "trace",
```

Comment:
* `-q`: only process errors
* ``: only process info messages or more severe, coming from flox, i.e. user facing logs
* `-v`: process debug messages or more severe, coming from anything flox maintained
* `-vv`:  process trace messages or more severe, coming from anything flox maintained
* `-vvv`:  process trace messages or more severe, coming from anything flox maintained, and debug messages of anything else
* `-vvvv...`: process traces messages of everything

These filters just influence which messages are being processed _at all_, not how.